### PR TITLE
Added PlanActorsController Service

### PR DESCRIPTION
**Pull request by abhijith0505**
This service maintains a database for the information of plan actors (rovers/astronauts).
The service allows adding into the database and retrieval from the database.

Docs: broke header fixes on Get-Descendant service's Readme

### CODE REVIEW

**Review by mtambos — COMMENTED** [review]
looks reasonable.
before merging into master, could you pass the code through flake8 and fix/purposefully ignore any errors and warnings?
also you might consider using an ORM for the database, so it's clear the data model you're proposing.
I like peewee. SQLAlchemy is another alternative.

**Comment by abhijith0505** [comment]
It has been passed through flake8. No errors/warnings.
So I should be using something like SQLAlchemy for the storage? Okay, shall do it

**Comment by mtambos** [comment]
> It has been passed through flake8. No errors/warnings.
weird. it should have given you a warning for missing an empty line before createDatabase, and also for that name not being snake_case

**Comment by abhijith0505** [comment]
Do you want me to make those changes explicitly?

**Comment by abhijith0505** [comment]
@mtambos Hello, have used an ORM now. Please check it

File: MOCC/src/configurations/PlanActorsController/src/plan_actor_schema.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+########################################################################
+
+
+class PlanActors(Base):
+    """"""
+    __tablename__ = "plan_actors"
+
+    address = Column(String, primary_key=True)
+    type = Column(String)
```
Comment (mtambos) [inline MOCC/src/configurations/PlanActorsController/src/plan_actor_schema.py]: "type" is a keyword from Python.
please change to something else

File: MOCC/src/configurations/PlanActorsController/src/DatabaseHelper.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# To add the plan actor fields into the database
+def add(data):
+    address = data[0]  # Address
+    type_t = data[1]  # Type
+    avail_start = data[2]  # available start date time
+    avail_end = data[3]  # available end date time
+    capabilities = data[4]  # capability string - JSON
+
+    query = s.query(PlanActors).filter(PlanActors.address.in_([address]))
+    result = query.first()
```
Comment (mtambos) [inline MOCC/src/configurations/PlanActorsController/src/DatabaseHelper.py]: please put inside a try-except-finally block.
as it is, if there's an exception, there'll be a non-commited transaction left open

**Comment by abhijith0505** [comment]
@mtambos Updated

File: MOCC/src/configurations/PlanActorsController/src/DatabaseHelper.py [hunk 3]
```diff
@@ -15,25 +15,29 @@
...
 
-    print("Inserted row successfully")
-    return 0
+            # commit the record the database
+            s.commit()
+            print("Inserted row successfully")
+            return 0
+
+    except:
+        return -1
```
Comment (mtambos) [inline MOCC/src/configurations/PlanActorsController/src/DatabaseHelper.py:40]: here you need to rollback the transaction, otherwise it remains open

Comment (abhijith0505) [reply MOCC/src/configurations/PlanActorsController/src/DatabaseHelper.py:40]: Please check

**Review by mtambos — APPROVED** [review]
good work.
it'd be good if you added some unit tests in the next pr

**Pull request merged**